skimage/color/_colorconv.pyx

```diff
@@ -0,0 +1,83 @@
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+
+import cython
+import numpy as np
+from libc.math cimport floor
+
+from .._shared.fused_numerics cimport np_floats
+
+
+def rgb2hsv_inner(np_floats[:, ::] rgb):
+    cdef:
+        np_floats[:, ::] hsv = np.empty_like(rgb)
+        np_floats delta, minv, maxv, tmp
+        Py_ssize_t i, n, ch
+
+    n = rgb.shape[0]
+    with nogil:
+        for i in range(n):
+            minv = maxv = rgb[i, 0]
+            for ch in range(1, 3):
+                tmp = rgb[i, ch]
+                if tmp > maxv:
+                    maxv = tmp
+                elif tmp < minv:
+                    minv = tmp
+            delta = maxv - minv
+            if delta == 0.0:
+                hsv[i, :2] = 0.0
+            else:
+                hsv[i, 1] = delta / maxv
+                if rgb[i, 0] == maxv:
+                    hsv[i, 0] = (rgb[i, 1] - rgb[i, 2]) / delta
+                elif rgb[i, 1] == maxv:
+                    hsv[i, 0] = 2.0 + (rgb[i, 2] - rgb[i, 0]) / delta
+                elif rgb[i, 2] == maxv:
+                    hsv[i, 0] = 4.0 + (rgb[i, 0] - rgb[i, 1]) / delta
+                hsv[i, 0] /= 6.0
+                hsv[i, 0] -= floor(<double>hsv[i, 0])
+            hsv[i, 2] = maxv
+    return np.asarray(hsv)
+
+
+def hsv2rgb_inner(np_floats[:, ::] hsv):
+    cdef:
+        np_floats[:, ::] rgb = np.empty_like(hsv)
+        np_floats f, v, p
+        Py_ssize_t i, n, hi, rem
+
+    n = rgb.shape[0]
+    with nogil:
```

Comment:
Are you able to add a reference for the implementation of the algorithm here? That would allow future people to continue to optimize it.

Reply:
yes, I have added references in 0fab7ef